[tesseract] Build error

**Host Environment**
- OS: macOS arm64
- Compiler: Apple clang version 13.0.0 (clang-1300.0.29.30)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install tesseract`

**Failure logs** 
`Error: tesseract[core] is only supported on '!arm'`

**Additional context**
It returns that tesseract is not supported on arm, even though using [official tesseract compilation guide](https://tesseract-ocr.github.io/tessdoc/Compiling.html#macos) you can compile tesseract with no problems on macos-arm64 (apart from specifying arm64 target as written in the guide).

You can try with `--allow-unsupported`, but for me it fails with 
```
CMake Error at cmake/OptimizeForArchitecture.cmake:78 (list):
  list index: 3 out of range (-1, 0)
Call Stack (most recent call first):
  CMakeLists.txt:129 (AutodetectHostArchitecture)
```
but it seems that this file was removed in version 5.0.1. So updating tesseract to the newest version should fix the error.

--- a/ports/tesseract/vcpkg.json
+++ b/ports/tesseract/vcpkg.json
@@ -1,11 +1,11 @@
 {
   "name": "tesseract",
   "version": "5.2.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "An OCR Engine that was developed at HP Labs between 1985 and 1995... and now at Google.",
   "homepage": "https://github.com/tesseract-ocr/tesseract",
   "license": "Apache-2.0",
-  "supports": "!(arm & (osx | linux))",
+  "supports": "!uwp",
   "dependencies": [
     "curl",
     "leptonica",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7046,7 +7046,7 @@
     },
     "tesseract": {
       "baseline": "5.2.0",
-      "port-version": 1
+      "port-version": 2
     },
     "tfhe": {
       "baseline": "1.0.1",

--- a/versions/t-/tesseract.json
+++ b/versions/t-/tesseract.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9d9b7d5ba2e222ee4fb62d4d5f03992e9232a97d",
+      "version": "5.2.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "ea93f36603ca265da43ef28d6583871ef3d97b43",
       "version": "5.2.0",
       "port-version": 1

